File: sol/solution/0001-0100/0034. Find First and Last Position of Element in Sorted Array/Solution.cs
public class Solution {
    public int[] SearchRange(int[] nums, int target) {
        return new int[] { FindBound(nums, target, true), FindBound(nums, target, false) };
    }
    private int FindBound(int[] nums, int target, bool first) {
        int l = 0, r = nums.Length - 1, ans = -1;
        while (l <= r) {
            int mid = l + (r - l) / 2;
            if (nums[mid] == target) {
                ans = mid;
                if (first) r = mid - 1;
                else l = mid + 1;
            } else if (nums[mid] < target) l = mid + 1;
            else r = mid - 1;
        }
        return ans;
    }
}

File: sol/solution/0001-0100/0034. Find First and Last Position of Element in Sorted Array/Solution.rs
impl Solution {
    pub fn search_range(nums: Vec<i32>, target: i32) -> Vec<i32> {
        fn find_bound(nums: &Vec<i32>, target: i32, first: bool) -> i32 {
            let (mut l, mut r) = (0, nums.len() as i32 - 1);
            let mut ans = -1;
            while l <= r {
                let mid = l + (r - l) / 2;
                if nums[mid as usize] == target {
                    ans = mid;
                    if first { r = mid - 1; } else { l = mid + 1; }
                } else if nums[mid as usize] < target {
                    l = mid + 1;
                } else {
                    r = mid - 1;
                }
            }
            ans
        }
        vec![find_bound(&nums, target, true), find_bound(&nums, target, false)]
    }
}